ftplugin/ledger.vim

```diff
@@ -398,12 +400,13 @@ unlet s:old s:new s:fun
 
 function! s:collect_completion_data() "{{{1
   let transactions = ledger#transactions()
-  let cache = {'descriptions': [], 'tags': {}, 'accounts': {}}
+  let cache = {'descriptions': [], 'tags': {}, 'accounts': {}, 'flat_accounts': []}
   if exists('g:ledger_accounts_cmd')
     let accounts = systemlist(g:ledger_accounts_cmd)
   else
     let accounts = ledger#declared_accounts()
   endif
+  let cache.flat_accounts = accounts
```

Comment:
Why do we need to cache two copies of the account list for this to work?

Reply:
`cache.account` is a nested structure whereas what we need for matchfuzzy is a list of full account names.

Reply:
That may be, but what you are assigning to the "flat" list is just the same account list we already have. I don't see where it is being flattened.

Reply:
They are indeed different representations:
<img width="1321" alt="Screenshot 2024-05-04 at 3 19 54 PM" src="https://github.com/ledger/vim-ledger/assets/7976315/1ff42054-b9ed-4ed2-9de8-4904151192e3">

Reply:
I believe this is where the nested structure is created: [ftplugin/ledger.vim#447](https://github.com/ledger/vim-ledger/blob/ca89864fa7452800276d9f2eb6abf9b6a61a0015/ftplugin/ledger.vim#L447-L453)

Reply:
Got it, I missed that connection. Thanks.